sound/soc/sof/intel/hda-sdw-bpt.c

```diff
@@ -104,8 +213,26 @@ static int hda_sdw_bpt_dma_enable(struct device *dev,
 	ret = hda_cl_trigger(sdev->dev, sdw_bpt_stream, SNDRV_PCM_TRIGGER_START);
 	if (ret < 0) {
 		dev_err(sdev->dev, "%s: SDW BPT DMA trigger start failed\n", __func__);
+
 		return ret;
 	}
+
+	if (!sdev->dspless_mode_selected) {
+		/* the chain DMA needs to be programmed before the DMAs */
+		ret = chain_dma_trigger(sdev,
+					hdac_stream(sdw_bpt_stream)->stream_tag,
+					hdac_stream(sdw_bpt_stream)->direction,
+					SOF_IPC4_PIPE_RUNNING);
+		if (ret < 0) {
```

Comment:
dont you need to do hda_cl_trigger STOP in this error path?

Reply:
> dont you need to do hda_cl_trigger STOP in this error path?

Yes, done.